### PR TITLE
Profile endpoint

### PR DESCRIPTION
**Pull request by williambaibai**
## Overview

- Modify `get_games_by_user.py` for the needs of our profile page

## Changes Made

- Sort by past games, my games, joined games in `get_games_by_user.py`

## Test Coverage

- tested locally, works fine

## Next Steps (delete if not applicable)

- Write API

### CODE REVIEW

File: src/app/controllers/get_all_games.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from . import *
 
+
```
Comment (alanna-zhou) [inline src/app/controllers/get_all_games.py:3]: was this added by black formatter?

File: src/app/controllers/get_games_by_user.py [hunk 2]
```diff
@@ -13,4 +14,17 @@ def content(self, **kwargs):
         user = users_dao.get_user_by_id(user_id)
 
         serialized_games = [game_dao.serialize_game(game) for game in user.games]
-        return utils.success_response(serialized_games)
+        result = {"my_games": [], "joined_games": [], "past_games": []}
+        for game in serialized_games:
+            game_time = utils.parse_datetime(game["time"])
```
Comment (alanna-zhou) [inline src/app/controllers/get_games_by_user.py:19]: game might not have "time"? game.get("time") might be safer?

File: src/app/controllers/get_games_by_user.py [hunk 3]
```diff
@@ -13,4 +14,17 @@ def content(self, **kwargs):
...
+            tz_info = game_time.tzinfo
+            if game_time < dt.now(tz=tz_info):
```
Comment (alanna-zhou) [inline src/app/controllers/get_games_by_user.py:21]: what if less than equal to?

File: src/utils.py [hunk 4]
```diff
@@ -6,11 +6,18 @@
...
 
+
 def failure_response(message, code=404):
     return json.dumps({"success": False, "error": message}), code
 
+
 def parse_time(time):
-    return dt.strptime(time, "%Y-%m-%d, %H:%M")
\ No newline at end of file
+    return dt.strptime(time, "%Y-%m-%d, %H:%M")
```
Comment (alanna-zhou) [inline src/utils.py:19]: might be more helpful for future backend devs if you refactor this into the string, and the string says what format this is (like is it epoch? -- ik it isn't but the formal date format name is super helpful)

File: src/utils.py [hunk 5]
```diff
@@ -6,11 +6,18 @@
...
+
+
+def parse_datetime(datetime):
+    return dt.strptime(datetime, "%Y-%m-%dT%H:%M:%S%z")
```
Comment (alanna-zhou) [inline src/utils.py:23]: same here

File: src/app/controllers/get_all_games.py (hunk 1 above)
Comment (williambaibai) [reply src/app/controllers/get_all_games.py:3]: Yup.

File: src/app/controllers/get_games_by_user.py (hunk 2 above)
Comment (williambaibai) [reply src/app/controllers/get_games_by_user.py:19]: game will always have this field

File: src/app/controllers/get_games_by_user.py (hunk 3 above)
Comment (williambaibai) [reply src/app/controllers/get_games_by_user.py:21]: sure

**Pull request merged**